### PR TITLE
Enhance audio processing with popup feedback and UI adjustments

### PR DESCRIPTION
**Pull request by ItsSimko**
@sourcery-ai review

### CODE REVIEW

File: src/FreeScribe.client/client.py [hunk 1]
```diff
@@ -260,15 +274,48 @@ def toggle_recording():
...
         DEFUALT_BUTTON_COLOUR = mic_button.cget('background')
         mic_button.config(bg="red", text="Stop\nRecording")
         start_flashing()
     else:
         is_recording = False
         if recording_thread.is_alive():
             recording_thread.join()  # Ensure the recording thread is terminated
+
+
+        popup = tk.Toplevel()
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:290]: We want a blocking call so it doesn't go to the PHI scribe

File: src/FreeScribe.client/client.py [hunk 2]
```diff
@@ -260,15 +274,48 @@ def toggle_recording():
...
+        # Disable closing of the popup manually
+        popup.protocol("WM_DELETE_WINDOW", lambda: None)
+
+        def animate_text():
+            # Create an animated text with moving ellipsis
+            current_text = popup_label.cget("text")
+            if current_text.endswith("..."):
+                popup_label.config(text="Processing Audio")
+            else:
+                popup_label.config(text=current_text + ".")
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:305]: This is fine.

File: src/FreeScribe.client/client.py [hunk 3]
```diff
@@ -249,7 +260,10 @@ def save_audio():
 DEFUALT_BUTTON_COLOUR = None
 
 def toggle_recording():
-    global is_recording, recording_thread, DEFUALT_BUTTON_COLOUR
+    global is_recording, recording_thread, DEFUALT_BUTTON_COLOUR, realtime_thread, audio_queue
+
+    realtime_thread = threaded_realtime_text()
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: Thread does clean up

Changed thread to only start if not recording: https://github.com/ClinicianFOCUS/FreeScribe/pull/45/commits/00f6b74b2fe7cdf31e6cb3706bd48a5960683a96

File: src/FreeScribe.client/client.py [hunk 4]
```diff
@@ -153,18 +156,26 @@ def record_audio():
             audio_buffer = np.frombuffer(data, dtype=np.int16).astype(np.float32) / 32768
             if is_silent(audio_buffer, app_settings.editable_settings["Silence cut-off"]):
                 silent_duration += CHUNK / RATE
+                record_duration += CHUNK / RATE
             else:
                 current_chunk.append(data)
+                record_duration += CHUNK / RATE
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/45/commits/a6eabba52a722d37f823455bb9a8859a7415340f

File: src/FreeScribe.client/client.py [hunk 5]
```diff
@@ -153,18 +156,26 @@ def record_audio():
...
-                    audio_queue.put(b''.join(current_chunk))
-                current_chunk = []
-                silent_duration = 0
+            if record_duration >= minimum_audio_duration:
+                if silent_duration >= minimum_silent_duration:
+                    if app_settings.editable_settings["Real Time"] and current_chunk:
+                        audio_queue.put(b''.join(current_chunk))
+                    current_chunk = []
+                    silent_duration = 0
+                    record_duration = 0
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/45/commits/2d619066dfb338fec45525fbce68400e41c5b154

**Review by yingbull — APPROVED** [review]

**Pull request merged**